File: frontend/src/services/authService.ts
import API from "../api/axios";

export interface RegisterData {
  full_name: string;
  email: string;
  password: string;
  phone: string;
  address: string;
  role: string;
}

export interface LoginData {
  email: string;
  password: string;
}

export const registerUser = async (data: RegisterData) => {
  const response = await API.post("/auth/register", data);
  return response.data;
};

export const loginUser = async (data: LoginData) => {
  const response = await API.post("/auth/login", data);
  return response.data;
};

File: backend/app/services/auth_service.py
from sqlalchemy.orm import Session

from app.models.user import User
from app.schemas.user_schema import UserRegister, UserLogin
from app.core.security import hash_password, verify_password


def register_user(user: UserRegister, db: Session):

    existing_user = (
        db.query(User)
        .filter(User.email == user.email)
        .first()
    )

    if existing_user:
        return None

    new_user = User(
        full_name=user.full_name,
        email=user.email,
        password=hash_password(user.password),
        phone=user.phone,
        address=user.address,
        role=user.role,
    )

    db.add(new_user)
    db.commit()
    db.refresh(new_user)

    return new_user


def login_user(user: UserLogin, db: Session):

    existing_user = (
        db.query(User)
        .filter(User.email == user.email)
        .first()
    )

    if existing_user is None:
        return None

    if not verify_password(user.password, existing_user.password):
        return None

    return existing_user